compat-2021/main.js

```diff
@@ -177,10 +204,14 @@ function scoreRuns(runs, allTestsSet) {
           subtestPasses = 1;
         }
 
-        if (subtestPasses == subtestTotal) {
-          passingTests += 1;
-        }
+        // A single test is scored based on how many of its subtests pass;
+        // these fractional values are then summed and normalized against the
+        // total number of tests to get the overall score.
+        passingTests += subtestPasses / subtestTotal;
 
+        // TODO: I suspect this doesn't handle missing test results properly,
```

Comment:
We are iterating the results of each run separately, and skipping anything not in `allTestsSet`. If there's a test we don't encounter in the results, then the array in `testResults.get(testname)` won't be pushed to.

If for some reason the results are missing only from some of the runs, that array might be too short, and we'd generate some weird CSV later in `csvRecord.push(result.join('/'))`.

So yeah, this is a good TODO to have. Feel free to copy in more detail if you think it's helpful.